semconv/resource.go

```diff
@@ -12,246 +12,734 @@
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'redis'
 	K8SContainerNameKey = attribute.Key("k8s.container.name")
+)
 
-	// The uid of the ReplicaSet.
-	K8SReplicaSetUIDKey = attribute.Key("k8s.replicaset.uid")
-
+// A Kubernetes ReplicaSet object.
+const (
+	// The UID of the ReplicaSet.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
+	K8SReplicasetUIDKey = attribute.Key("k8s.replicaset.uid")
 	// The name of the ReplicaSet.
-	K8SReplicaSetNameKey = attribute.Key("k8s.replicaset.name")
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'opentelemetry'
+	K8SReplicasetNameKey = attribute.Key("k8s.replicaset.name")
+)
 
-	// The uid of the Deployment.
+// A Kubernetes Deployment object.
+const (
+	// The UID of the Deployment.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
 	K8SDeploymentUIDKey = attribute.Key("k8s.deployment.uid")
-
-	// The name of the deployment.
+	// The name of the Deployment.
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'opentelemetry'
 	K8SDeploymentNameKey = attribute.Key("k8s.deployment.name")
+)
 
-	// The uid of the StatefulSet.
-	K8SStatefulSetUIDKey = attribute.Key("k8s.statefulset.uid")
-
+// A Kubernetes StatefulSet object.
+const (
+	// The UID of the StatefulSet.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
+	K8SStatefulsetUIDKey = attribute.Key("k8s.statefulset.uid")
 	// The name of the StatefulSet.
-	K8SStatefulSetNameKey = attribute.Key("k8s.statefulset.name")
-
-	// The uid of the DaemonSet.
-	K8SDaemonSetUIDKey = attribute.Key("k8s.daemonset.uid")
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'opentelemetry'
+	K8SStatefulsetNameKey = attribute.Key("k8s.statefulset.name")
+)
 
+// A Kubernetes DaemonSet object.
+const (
+	// The UID of the DaemonSet.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
+	K8SDaemonsetUIDKey = attribute.Key("k8s.daemonset.uid")
 	// The name of the DaemonSet.
-	K8SDaemonSetNameKey = attribute.Key("k8s.daemonset.name")
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'opentelemetry'
+	K8SDaemonsetNameKey = attribute.Key("k8s.daemonset.name")
+)
 
-	// The uid of the Job.
+// A Kubernetes Job object.
+const (
+	// The UID of the Job.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
 	K8SJobUIDKey = attribute.Key("k8s.job.uid")
-
 	// The name of the Job.
+	//
+	// Type: string
+	// Required: No
+	// Examples: 'opentelemetry'
 	K8SJobNameKey = attribute.Key("k8s.job.name")
+)
 
-	// The uid of the CronJob.
-	K8SCronJobUIDKey = attribute.Key("k8s.cronjob.uid")
-
+// A Kubernetes CronJob object.
+const (
+	// The UID of the CronJob.
+	//
+	// Type: string
+	// Required: No
+	// Examples: '275ecb36-5aa8-4c2a-9c47-d8bb681b9aff'
+	K8SCronjobUIDKey = attribute.Key("k8s.cronjob.uid")
```

Comment:
Can we add an override here to capitalize the two worded "cron job"?

```suggestion
	K8SCronJobUIDKey = attribute.Key("k8s.cronjob.uid")
```